### DISCUSSION TITLE
Aws codecommit support

### DISCUSSION THREAD

**Discussion by vinuthahp** [fluxcd/flux2#1413] [General]
Does fluxV2 bootstrapping or flux install command supports to register AWS codecommit repo?

Getting an error while creating Kustomization:
below  is the error:

Bastion : model-d967651-eks :~$ ./flux create kustomization test  --source=GitRepository/test  --path="./clusters"   --prune=true --interval=15m
✚ generating Kustomization
► applying Kustomization
✔ Kustomization updated
◎ waiting for Kustomization reconciliation
**✗ context deadline exceeded**

Any fix for this?

**Comment by kingdonb** [comment]
Can you provide the full `flux create source git` command you used to populate the cluster with `GitRepository/test`, or the yaml for that generated resource as well?

What is the output of `flux get sources git` please, and have you already tried setting `--git-implementation [xxx]` to both possible values, `libgit2` and `go-git`

We have received reports before from AWS codecommit users that Flux doesn't work there, but I have not yet been able to work with someone directly and see what is happening to result in this type of failure exactly. I suspect if you probe for more information about the errors in Flux, you're going to find authentication failures and I don't want to speculate further until I know more about it...

Could you please also include the output of `kubectl describe -n flux-system gitrepository test`, (it's a long output, I'm only specifically interested in the events and status informations nearer towards the bottom of the listing)

**Reply by vinuthahp** [reply]
Please find the requested details below:
flux create source git test
  --git-implementation=libgit2    
  --url=ssh://<ssh-key-id>@git-codecommit.amazonaws.com/v1/repos/test
  --branch=model-test-eks  
  --ssh-key-algorithm=rsa     
  --ssh-rsa-bits=4096  
  --interval=1m --private-key-file=/home/test-eks/.ssh/id_rsa


 
 model-test-eks :~$ .**/flux get sources git**
NAME            READY   MESSAGE                                                                         REVISION                        SUSPENDED
test      True    Fetched revision: model-test-eks/3f42bf9de601b1675638336a993c4f93d9ba96cc33    model-test-eks/3f42bf9de601b1715eytr74r888993c4f93d9ba96cc33       False
testcluster     True    Fetched revision: model-test-eks/3f42bf9de601b1715c6782ea993c4f93d9ba96cc33    model-test-eks/3f42bf967ft25dd601b1715c436a993c4f93d9ba96cc33       False


I haven't set any values for --git-implementation

Here is the status and events : kubectl describe -n flux-system gitrepository test

Status:
  Artifact:
    Checksum:          e8af40b5ece55a66f0103fb7246bc99f75115244
    Last Update Time:  2021-05-16T15:45:39Z
    Path:              gitrepository/flux-system/testcluster/3f42bf9de601b1715c436a993c4f93d9ba96cc33.tar.gz
    Revision:          model-test-eks/3f42bf9de601b1715c436a993c4f93d9ba96cc33
    URL:               http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/testcluster/3f42bde601b1715c436a993c4f93d9ba96cc33.tar.gz
  Conditions:
    Last Transition Time:  2021-05-16T15:45:39Z
    Message:               Fetched revision: model-d967651-eks/3f42bf9de601b1715c436a993c4f93d9ba96cc33
    Reason:                GitOperationSucceed
    Status:                True
    Type:                  Ready
  Observed Generation:     1
  URL:                     http://source-controller.flux-system.svc.cluster.local./gitrepository/flux-system/testcluster/latest.tar.gz
Events:
  Type    Reason  Age                  From               Message
  ----    ------  ----                 ----               -------
  Normal  error   20m (x47 over 3d8h)  source-controller  unable to clone 'ssh://<ssh-key-id>@git-codecommit.ap-southeast-2.amazonaws.com/v1/repos/test', error: SSH could not write data: Failure while draining incoming flow

**Reply by kingdonb** [reply]
> I haven't set any values for --git-implementation

It looks like you have actually,

```
--git-implementation=libgit2
```

Do you have better results if you swap that out for `go-git`?

**Reply by vinuthahp** [reply]
Tried with go-git git-implementation method , still getting the same error.
could you please let me know if there are any suggestion for codecommit as source.

**Reply by vinuthahp** [reply]
Team, Any update on the above mentioned issue

**Reply by phillebaba** [reply]
Hello,

My guess is that the reason you are having issues is because you have not set the correct user, as it seems like AWS requires the correct ID to be used. Have a look at the docs where it mentions SSH Key ID.
https://docs.aws.amazon.com/codecommit/latest/userguide/setting-up-ssh-unixes.html

Could you try adding the ID to you URL, it shoudl be picked up by Source Controller.
